packages/flutter/lib/src/material/page.dart

```diff
@@ -105,6 +116,18 @@ mixin MaterialRouteTransitionMixin<T> on PageRoute<T> {
     return themeDelegatedTransition != null ? themeDelegatedTransition(context, animation, secondaryAnimation, allowSnapshotting, child) : null;
   }
 
+  @override
+  TickerFuture didPush() {
+    controller?.duration = transitionDuration;
```

Comment:
not in this class, I meant in TransitionRoute class directly
https://github.com/flutter/flutter/blob/d7f5547ecd036d80f408935e4dc154a0dbe0b8d8/packages/flutter/lib/src/widgets/routes.dart#L291